src/ws/server.rs

```diff
@@ -432,22 +434,25 @@ async fn background_task(mut server: RawServer, mut from_front: mpsc::UnboundedR
 						active_subscriptions.insert(sub_id, *sub_unique_id);
 					}
 				} else if let Some(sub_unique_id) = unsubscribe_methods.get(request.method()) {
-					if let Ok(sub_id) = RawServerSubscriptionId::from_wire_message(&JsonValue::Null) {
-						// FIXME: from request params
-						debug_assert!(subscribed_clients.contains_key(&sub_unique_id));
-						if let Some(clients) = subscribed_clients.get_mut(&sub_unique_id) {
-							// TODO: we don't actually check whether the unsubscribe comes from the right
-							//       clients, but since this the ID is randomly-generated, it should be
-							//       fine
-							if let Some(client_pos) = clients.iter().position(|c| *c == sub_id) {
-								clients.remove(client_pos);
-							}
-
-							if let Some(s_u_id) = active_subscriptions.remove(&sub_id) {
-								debug_assert_eq!(s_u_id, *sub_unique_id);
-							}
-						}
-					}
+					log::debug!("[backend]: server received unsubscription: {:?}", request);
+					match RawServerSubscriptionId::try_from(request.params()) {
+                        Ok(sub_id) => {
+                            debug_assert!(subscribed_clients.contains_key(&sub_unique_id));
+                            if let Some(clients) = subscribed_clients.get_mut(&sub_unique_id) {
+                                // TODO: we don't actually check whether the unsubscribe comes from the right
+                                //       clients, but since this the ID is randomly-generated, it should be
+                                //       fine
+                                if let Some(client_pos) = clients.iter().position(|c| *c == sub_id) {
+                                    clients.remove(client_pos);
+                                }
+
+                                if let Some(s_u_id) = active_subscriptions.remove(&sub_id) {
+                                    debug_assert_eq!(s_u_id, *sub_unique_id);
+                                }
+                            }
+                        }
+                        Err(_) => log::error!("Unsubscription of method=\"{}\" failed; The subscription ID must passed as the first argument of Array or \"subscription\" name of Object, got={:?}", request.method(), request.params()),
```

Comment:
Indentiation is off (tabs vs sapces), otherwise looks fine.

Reply:
nice catch, I must have messed it up during merging but still weird that rustfmt didn't fix it.